app/assets/src/components/Samples.jsx

```diff
@@ -1708,6 +1712,125 @@ function TableDownloadDropdown({ project_id, parent }) {
   );
 }
 
+class BackgroundModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalOpen: false,
+      name: "",
+      description: "",
+      sample_names: []
+    };
+    this.handleOpen = this.handleOpen.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.renderTextField = this.renderTextField.bind(this);
+    this.sample_ids = props.parent.state.selectedSampleIds;
+  }
+  handleOpen() {
+    axios.get(`/show_sample_names?sample_ids=${this.sample_ids}`).then(res => {
+      this.setState({
+        modalOpen: true,
+        name: "",
+        description: "",
+        sample_names: res.data.sample_names
+      });
+    });
+    this.props.parent.setState({
+      background_creation_response: {}
+    });
+  }
+  handleClose() {
+    this.setState({
+      modalOpen: false,
+      name: "",
+      description: "",
+      sample_names: []
+    });
+  }
+  handleChange(e, { name, value }) {
+    this.setState({ [e.target.id]: value });
+  }
+  handleSubmit() {
+    this.props.parent.handleCreateBackground(
+      this.state.new_background_name,
+      this.state.new_background_description,
+      this.props.parent.state.selectedSampleIds
+    );
+  }
+  renderTextField(label, id) {
+    return (
+      <Form.Field>
+        <Form.Input
+          label={label}
+          placeholder={label}
+          className="col s12 browser-default"
+          id={id}
+          onChange={this.handleChange}
+        />
+      </Form.Field>
+    );
+  }
+
+  render() {
+    let background_creation_response = this.props.parent.state
+      .background_creation_response;
+    return (
+      <Modal
+        trigger={
+          <CohortButton
```

Comment:
Do we use the same button as compare? Or is this temporary until you get feedback from Jenn?

Reply:
Yes to both :D Conceptually it should be a similar button because it represents an action on a group of samples. But it is also temporary until I get a design from Jenn

Reply:
I believe the icon makes not sense for this particular action.

Reply:
ok removed icon